pkg/ipam/allocator/multipool/allocator.go

```diff
@@ -5,131 +5,66 @@ package multipool
 
 import (
 	"context"
-	"errors"
-	"fmt"
-	"strconv"
-	"strings"
 
 	"github.com/sirupsen/logrus"
 
-	operatorOption "github.com/cilium/cilium/operator/option"
 	"github.com/cilium/cilium/pkg/ipam"
 	"github.com/cilium/cilium/pkg/ipam/allocator"
 	cilium_v2alpha1 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1"
+	"github.com/cilium/cilium/pkg/logging"
+	"github.com/cilium/cilium/pkg/logging/logfields"
 )
 
+var log = logging.DefaultLogger.WithField(logfields.LogSubsys, "ipam-allocator-multi-pool")
+
 // Allocator implements allocator.AllocatorProvider
 type Allocator struct {
 	poolAlloc *PoolAllocator
 }
 
-type poolSpec struct {
-	ipv4CIDRs    []string
-	ipv6CIDRs    []string
-	ipv4MaskSize int
-	ipv6MaskSize int
+func (a *Allocator) Init(ctx context.Context) error {
+	a.poolAlloc = NewPoolAllocator()
+	return nil
 }
 
-const (
-	poolKeyIPv4CIDRs    = "ipv4-cidrs"
-	poolKeyIPv4MaskSize = "ipv4-mask-size"
-	poolKeyIPv6CIDRs    = "ipv6-cidrs"
-	poolKeyIPv6MaskSize = "ipv6-mask-size"
-)
-
-// parsePoolSpec parses a pool spec string in the form
-// "ipv4-cidrs:172.16.0.0/16,172.17.0.0/16;ipv4-mask-size:24".
-func parsePoolSpec(poolString string) (poolSpec, error) {
-	fields := strings.Split(strings.ReplaceAll(poolString, " ", ""), ";")
-
-	var pool poolSpec
```

Comment:
GitHub says (on the diff for the last commit):

![image](https://github.com/cilium/cilium/assets/17001771/6adc4082-7eaf-4aff-8d30-4abd191dc787)